File: CORREDOR/4 Lenguajes/Java/Atributos.java
package Java;

    class Atributos{
        public String nombre;
        private int edad;


        public Atributos(String nombre, int edad) {
            this.nombre = nombre;
            this.edad = edad;
        }

        public void caminar(){
            System.out.println( nombre + " esta caminando en el parque y tiene " + edad + " años");
        }
        public static void main(String[] args) {

        Atributos p1 = new Atributos("Johan", 18);
        p1.caminar();

        Coche c1 = new Coche("Audi", "rojo", 300);
        c1.acelerar();

        Libro l1 = new Libro("Don quijote", "Gabriel Garcia", 280);
        l1.mostrarInfo();

        Celular c2 = new Celular("Samsung", "15", 1500000);
        c2.mostrarDatos();
        }
    }

    class Coche{
        public String marca;
        private String color;
        public int velocidad;

        public Coche(String marca, String color, int velocidad) {
            this.marca = marca;
            this.color = color;
            this.velocidad = velocidad;
        }

        public void acelerar(){
            System.out.println("El " + marca + " de color " + color + " va a " + velocidad + "km/h");
        }
    }

    class Libro {
    public String titulo;
    private String autor;
    public int paginas;

    public Libro(String titulo, String autor, int paginas) {
        this.titulo = titulo;
        this.autor = autor;
        this.paginas = paginas;
    }

    public void mostrarInfo() {
        System.out.println("El libro '" + titulo + "' fue escrito por " + autor + " y tiene " + paginas + " páginas.");
    }
}

    class Celular {
    public String marca;
    private String modelo;
    public double precio;

    public Celular(String marca, String modelo, double precio) {
        this.marca = marca;
        this.modelo = modelo;
        this.precio = precio;
    }

    public void mostrarDatos() {
        System.out.println("Celular " + marca + " modelo " + modelo + " cuesta $" + precio);
    }
}

File: CORREDOR/4 Lenguajes/Go/Atributos.go
package main

import "fmt"

type Atributos struct {
	Nombre string
	edad   int
}

func NewAtributos(nombre string, edad int) Atributos {
	return Atributos{Nombre: nombre, edad: edad}
}

func (a Atributos) Caminar() {
	fmt.Printf("%s está caminando en el parque y tiene %d años\n", a.Nombre, a.edad)
}

type Coche struct {
	Marca     string
	color     string
	Velocidad int
}

func NewCoche(marca string, color string, velocidad int) Coche {
	return Coche{Marca: marca, color: color, Velocidad: velocidad}
}

func (c Coche) Acelerar() {
	fmt.Printf("El %s de color %s va a %d km/h\n", c.Marca, c.color, c.Velocidad)
}

type Libro struct {
	Titulo  string
	autor   string
	Paginas int
}

func NewLibro(titulo string, autor string, paginas int) Libro {
	return Libro{Titulo: titulo, autor: autor, Paginas: paginas}
}

func (l Libro) MostrarInfo() {
	fmt.Printf("El libro '%s' fue escrito por %s y tiene %d páginas.\n", l.Titulo, l.autor, l.Paginas)
}

type Celular struct {
	Marca  string
	modelo string
	Precio float64
}

func NewCelular(marca string, modelo string, precio float64) Celular {
	return Celular{Marca: marca, modelo: modelo, Precio: precio}
}

func (c Celular) MostrarDatos() {
	fmt.Printf("Celular %s modelo %s cuesta $%.2f\n", c.Marca, c.modelo, c.Precio)
}

func main() {
	p1 := NewAtributos("Johan", 18)
	p1.Caminar()

	c1 := NewCoche("Audi", "rojo", 300)
	c1.Acelerar()

	l1 := NewLibro("Don Quijote", "Gabriel Garcia", 280)
	l1.MostrarInfo()

	c2 := NewCelular("Samsung", "15", 1500000)
	c2.MostrarDatos()
}